lib/compeon/api_client.rb

```diff
@@ -64,8 +76,23 @@ def deep_transform_keys(object, &block)
       end
     end
 
-    class JSONAPIMiddleware < Faraday::Response::Middleware
-      Faraday::Response.register_middleware jsonapi: self
+    class JSONAPIRequestMiddleware < Faraday::Response::Middleware
+      Faraday::Request.register_middleware jsonapi_request: self
+
+      def call(env)
+        if env.body
+          body = DeepHashTransformer.deep_transform_keys(env.body) do |key|
+            key.to_s.tr('_', '-')
+          end
+
+          env.body = JSON.pretty_generate(body)
```

Comment:
I don't know if I like the pretty generated approach. What's the benefit for the big overhead in size?

Reply:
I think size isn't a real overhead. The benefit is better log readability.

Reply:
I'm not sure if that justifies increasing the payload size by ~50% for each request. Maybe (error) logging tools should be able to handle this. Rollbar already does this.